samples/core/parameterized_tfx_oss/taxi_pipeline_notebook.ipynb

```diff
@@ -0,0 +1,343 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# TFX pipeline example - Chicago Taxi tips prediction\n",
+    "\n",
+    "## Overview\n",
+    "[Tensorflow Extended (TFX)](https://github.com/tensorflow/tfx) is a Google-production-scale machine\n",
+    "learning platform based on TensorFlow. It provides a configuration framework to express ML pipelines\n",
+    "consisting of TFX components, which brings the user large-scale ML task orchestration, artifact lineage, as well as the power of various [TFX libraries](https://www.tensorflow.org/resources/libraries-extensions). Kubeflow Pipelines can be used as the orchestrator supporting the \n",
+    "execution of a TFX pipeline.\n",
+    "\n",
+    "This sample demonstrates how to author a ML pipeline in TFX and run it on a KFP deployment. \n",
+    "\n",
+    "## Permission\n",
+    "\n",
+    "This pipeline requires Google Cloud Storage permission to run. \n",
+    "If KFP was deployed through K8S marketplace, please follow instructions in [the guideline](https://github.com/kubeflow/pipelines/blob/master/manifests/gcp_marketplace/guide.md#gcp-service-account-credentials)\n",
+    "to make sure the service account has `storage.admin` role."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python3 -m pip install pip --upgrade --quiet --user\n",
+    "!python3 -m pip install kfp --upgrade --quiet --user\n",
+    "!python3 -m pip install tfx --upgrade --quiet --user"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example we'll need a very recent version of TFX SDK to leverage the [`RuntimeParameter`](https://github.com/tensorflow/tfx/blob/93ea0b4eda5a6000a07a1e93d93a26441094b6f5/tfx/orchestration/data_types.py#L137) feature.\n",
+    "\n",
+    "## RuntimeParameter in TFX DSL\n",
+    "Currently, TFX DSL only supports parameterizing field in the `PARAMETERS` section of `ComponentSpec`, see [here](https://github.com/tensorflow/tfx/blob/93ea0b4eda5a6000a07a1e93d93a26441094b6f5/tfx/types/component_spec.py#L126). This prevents runtime-parameterizing the pipeline topology. Also, if the declared type of the field is a protobuf, the user needs to pass in a dictionary with exactly the same names for each field, and specify one or more value as `RuntimeParameter` objects."
```

Comment:
>the user needs to pass in a dictionary with exactly the same names for each field

We can mention the`google.protobuf.json_format.MessageToDict` method which produces the compatible dictionaries.

Reply:
Done. PTAL